client/src/components/directory/UserDialog.jsx

```diff
@@ -115,7 +114,7 @@ class UserDialog extends Component {
   }
 
   onClickSubmit = async () => {
-    const { actions, mode, selectedUser, onClickCloseDialog } = this.props;
+    const { actions, mode, selectedUser, onClickCloseDialog, errorMessage } = this.props;
```

Comment:
Where does `errorMessage` come from? I thought it's named `directoryErrorMessage` in `App.jsx`

Reply:
True, guess we never tested unhappy path. Will change that!

Reply:
Then should it be error after all?

Reply:
I'm not sure what Martin intends to do...I think passing in the error from App takes a lot of work - are we supposed to have one prop for each page's errors? We can refactor to have an errorReducer handle all of this, but at this point it's a little late for such a big change.

Personally I would just check the response for errors. If there's an error in this case, doing `const response = await actions.addUser()` means that the response will contain a prop named `errorMessage` (although I think it's better to name it `error`, I explained the rationale in another comment).

Reply:
Its absolutely fine to use the error from the response. So instead of returning we fire the error

Reply:
Okay wait so we're keeping both approaches?

Reply:
No, i've changed it to an error object with a message inside

Reply:
so the message will be response.error.message

Reply:
So why do we still need to add errorMessage to the directoryReducer?

Reply:
Nevermind it's not there anymore, looks good to me now! Thanks :)